contracts/Compass.vy

```diff
@@ -173,51 +233,58 @@ def update_valset(consensus: Consensus, new_valset: Valset):
 # This makes calls to contracts that execute arbitrary logic
 # message_id is to prevent replay attack and every message_id can be used only once
 @external
-def submit_logic_call(consensus: Consensus, args: LogicCallArgs, message_id: uint256, deadline: uint256):
-    assert block.timestamp <= deadline, "Timeout"
+def submit_logic_call(consensus: Consensus, args: LogicCallArgs, fee_args: FeeArgs, message_id: uint256, deadline: uint256):
```

Comment:
We need another change, here is why:

At the moment, every Pigeon will run `estimate_gas` with this transaction data in order to get the amount of gas needed to run this TX. **BUT**, we have the `msg.sender` in the checkpoint (to make sure the message was actually sent by the intended relayer to avoid front running). Of course, this will fail for every Pigeon but the intended relayer, and they will get an incorrect amount of gas.

We can solve this issue by adding a new argument to `submit_logic_call`, `update_valset` and `submit_batch`. This new argument should be `intended_relayer:address`. Compass then needs to verify this argument is actually in the consensus (instead of `msg.sender`). It can then proceed and use this argument to distribute rewards instead of `msg.sender`.

In other words, we no longer care WHO actually sent the message. The protocol will ensure that only the intended relayer actually ends up sending the TX. And since the `msg.sender` no longer receives any rewards, front running is not possible either.